Author: Emmanuel Kasper <emmanuel@libera.cc>, Cesare Falco <cesare.falco@gmail.com>
Description: Add new makefile switch to allow the use of jpeg system headers
Forwarded: http://forums.bannister.org/ubbthreads.php?ubb=showflat&Number=76614&page=3

Index: mame/makefile
===================================================================
--- mame.orig/makefile	2012-02-26 22:21:48.000000000 +0100
+++ mame/makefile	2012-02-26 22:22:36.000000000 +0100
@@ -214,6 +214,9 @@
 # uncomment next line to include the symbols
 # SYMBOLS = 1
 
+# uncomment next line to build jpeglib as part of MAME build
+BUILD_JPEGLIB = 1
+
 # specify symbols level or leave commented to use the default
 # (default is SYMLEVEL = 2 normally; use 1 if you only need backtrace)
 # SYMLEVEL = 2
@@ -628,7 +631,14 @@
 # add formats emulation library
 FORMATS_LIB = $(OBJ)/libformats.a
 
+# add jpeglib image library
+ifeq ($(BUILD_JPEGLIB),1)
+INCPATH += -I$(SRC)/lib/libjpeg
 JPEG_LIB = $(OBJ)/libjpeg.a
+else
+LIBS += -ljpeg
+JPEG_LIB =
+endif
 
 #-------------------------------------------------
 # 'default' target needs to go here, before the 
Index: mame/src/mame/drivers/sliver.c
===================================================================
--- mame.orig/src/mame/drivers/sliver.c	2012-02-26 22:21:48.000000000 +0100
+++ mame/src/mame/drivers/sliver.c	2012-02-26 22:21:53.000000000 +0100
@@ -67,7 +67,7 @@
 #include "sound/okim6295.h"
 #include "cpu/mcs51/mcs51.h"
 #include "video/ramdac.h"
-#include "../../lib/libjpeg/jpeglib.h"
+#include <jpeglib.h>
 
 
 #define FIFO_SIZE 1024
Index: mame/src/osd/osdcomm.h
===================================================================
--- mame.orig/src/osd/osdcomm.h	2012-02-26 22:21:48.000000000 +0100
+++ mame/src/osd/osdcomm.h	2012-02-26 22:21:53.000000000 +0100
@@ -143,6 +143,10 @@
 
 #endif
 
+/* This should prevent libjpeg of typedef'ing INT32 */
+#ifndef XMD_H
+#define XMD_H
+#endif
 
 
 /***************************************************************************
